src/e2e/helpers/mobile/gestures/gesture.ts

```diff
@@ -1,37 +1,37 @@
 import { Direction } from '../../../../types'
-import { Browser } from 'webdriverio'
+import { Browser, TouchAction } from 'webdriverio'
 
 export interface GestureOptions {
   xStart?: number,
   yStart?: number,
   segmentLength?: number,
 }
 
-interface Action<T = any> {
-  action: string,
-  options?: T,
-}
+// Appium needs wait action before 'move' actions. If we don't add 'wait' actions none of the touch event is triggered.
+const WAIT_ACTION = { action: 'wait', ms: 50 } as TouchAction
 
 /** Apply gesture action for the given path. */
 const gesture = async (browser: Browser<'async'>, path: Direction[], { xStart = 70, yStart = 300, segmentLength = 70 }: GestureOptions = {}) => {
 
-  const moveActions = path.reduce<Action[]>((acc, cur) => {
-    const { options: { x: previousX, y: previousY } } = acc.length > 0
+  const moveActions = path.reduce<TouchAction[]>((acc, cur) => {
+    const { x: previousX, y: previousY } = acc.length > 0
       ? acc[acc.length - 1]
-      : { options: { x: xStart, y: yStart } }
-    const x = previousX + (cur === 'r' ? +segmentLength : cur === 'l' ? -segmentLength : 0)
-    const y = previousY + (cur === 'd' ? +segmentLength : cur === 'u' ? -segmentLength : 0)
-    return [...acc, { action: 'moveTo', options: { x, y } }]
+      : { x: xStart, y: yStart }
+    const x = previousX! + (cur === 'r' ? +segmentLength : cur === 'l' ? -segmentLength : 0)
+    const y = previousY! + (cur === 'd' ? +segmentLength : cur === 'u' ? -segmentLength : 0)
+    return [...acc, WAIT_ACTION, { action: 'moveTo', x, y }]
   }, [])
 
   // add first and last action
   const actions = [
-    { action: 'press', options: { x: xStart, y: yStart } },
+    { action: 'press', x: xStart, y: yStart },
     ...moveActions,
     { action: 'release' }
-  ]
+  ] as TouchAction[]
 
-  await browser.touchPerform(actions)
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  await browser.touchAction(actions)
```

Comment:
As I can see, webdriverio has a type issue in touchAction function. Can you confirm that, please? @raineorshine

Reply:
`browser.touchAction` performs a single action. Do you mean `touchPerform`?

> 'Touch Perform' works similarly to the other singular touch interactions, except that this allows you to chain together more than one touch action as one command. This is useful because Appium commands are sent over the network and there's latency between commands. This latency can make certain touch interactions impossible because some interactions need to be performed in one sequence. Vertical, for example, requires pressing down, moving to a different y coordinate, and then releasing. For it to work, there can't be a delay between the interactions.

https://appium.io/docs/en/commands/interactions/touch/touch-perform/

Reply:
🤔
Here is webdriverio documentation: 
https://webdriver.io/docs/api/browser/touchAction/

Here is an example from this document: 
```    
// multi action on an element
// drag&drop from position 200x200 down 100px on the screen
  browser.touchAction([
      { action: 'press', x: 200, y: 200 },
      { action: 'moveTo', x: 200, y: 300 },
      'release'
  ])```

I will look into the details of the touchPerform.